site/static/md/reference/attribute_plugins.md

```diff
@@ -110,6 +110,22 @@ Computed signals are useful for memoizing expressions containing other signals.
 <div data-text="$foo"></div>
 ```
 
+`data-computed` is a pure reactive function, this has several implications:
+
+1. If a computed signal is not consumed, then the computation will not execute
```

Comment:
Are you sure about this? It doesn’t sound right to me.

Reply:
Yes it is accurate. Because computed signals are pure functions, they cannot have side effects, which means they can be lazily computed.

This is how I originally discovered that `data-computed` is not to be used for side effects. I wanted to do a side-effect in a computed signal that otherwise wasn't used. See the  [discord conversation](https://discord.com/channels/1296224603642925098/1333801016642244739/1351641628997324844) where @delaneyj  very pointedly makes me aware of that fact.

Anyways you can try it out for yourself, define a `data-computed-my-computation="console.log('foobar') && 'woot'"` Then if you do not use the `$myComputation` signal, the log will not fire. but if you do `data-text="$myComputation"` you'll see the text is `woot` and a console log fires.

Reply:
I tested it and see `foobar` in the console, as expected. Please retest.

Reply:
Yes you’re right. Unaware, I was testing using a template that contained `ctx.signals.JSON()`.